### PR TITLE
Store locale preference 

### PR DESCRIPTION
**Pull request by hochiw**
- Introduced useLocalStorage hook
- Added context to appLocaleProvider for locale setting
- Parse local storage for locale preference 
- Added a dummy button in HomePage for switching locale

refs #18

### CODE REVIEW

File: react-app/src/i18n/translations/zh.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "App.title": "你好世界"
```
Comment (chihimng) [inline react-app/src/i18n/translations/zh.json:2]: :sosad:

File: react-app/src/providers/AppLocaleProvider.tsx [hunk 2]
```diff
@@ -9,36 +9,69 @@ import {
...
+);
+
 const AppLocaleProvider: React.FC<AppLocaleProviderProps> = (props) => {
   const { children } = props;
 
-  // TODO: keep user local settings
-  const [locale, _setLocale] = useState<Locale>(Locale.en);
+  const [localePref, setLocalePref] =
+    useLocalStorage<Locale>(LOCALE_STORAGE_KEY);
+  const [appLocale, setAppLocale] = useState<Locale>(localePref ?? Locale.en);
```
Comment (chihimng) [inline react-app/src/providers/AppLocaleProvider.tsx:33]: keeping two states looks a little weird to me 🤔

File: react-app/src/providers/AppLocaleProvider.tsx [hunk 3]
```diff
@@ -9,36 +9,69 @@ import {
...
+
   const translation = useMemo(() => {
-    return Translations[locale];
-  }, [locale]);
+    return Translations[appLocale];
```
Comment (chihimng) [inline react-app/src/providers/AppLocaleProvider.tsx:36]: hypothetically, a user can edit the local storage values in developer menu and cause error here? should we sanitize the value?

File: react-app/src/providers/AppLocaleProvider.tsx (hunk 2 above)
Comment (chihimng) [reply react-app/src/providers/AppLocaleProvider.tsx:33]: should we handle default value fallback at the useMemo instead? (along with checking if the local is valid, see below)

File: react-app/src/components/HomeScreen/HomeScreen.tsx [hunk 4]
```diff
@@ -1,15 +1,35 @@
...
 import { Link } from "react-router-dom";
 import AppRoutes from "../../navigation/AppRoutes";
 import LocalizedText from "../Localized/LocalizedText";
+import { useLocale } from "../../providers/AppLocaleProvider";
+import { Locale } from "../../i18n/LocaleModel";
 
 const HomeScreen: React.FC = () => {
+  const { setLocale: setLocale_ } = useLocale();
+
+  const setLocale = useCallback(
```
Comment (chihimng) [inline react-app/src/components/HomeScreen/HomeScreen.tsx:12]: onClick callback should handle React.MouseEvent...
please refactor the button component?

**Review by chihimng — CHANGES_REQUESTED** [review]

**Comment by hochiw** [comment]
@chihimng updated 🙏

**Review by chihimng — APPROVED** [review]

**Pull request merged**